File: src/main/java/com/example/accountingofsocks/repository/SocksDao.java
package com.example.accountingofsocks.repository;

import com.example.accountingofsocks.model.Socks;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.List;

@Repository
public interface SocksDao extends JpaRepository<Socks,Long> {

    List<Socks> findAllByColorAndCottonPartGreaterThan(String color,byte cottonPart);

    List<Socks> findAllByColorAndCottonPartEquals(String color,byte cottonPart);

    List<Socks> findAllByColorAndCottonPartIsLessThan(String color,byte cottonPart);
}

File: src/main/kotlin/com/projects/raifservice/dao/SocksDao.kt
package com.projects.raifservice.dao

import com.projects.raifservice.model.Socks
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface SocksDao : CrudRepository<Socks, Long> {

    fun findSocksByColorAndCottonPart (color: String, cottonPart: Int): Socks?

    fun findSocksByColorAndCottonPartIsGreaterThan (color: String, cottonPart: Int): List<Socks>

    fun findSocksByColorAndCottonPartIsLessThan (color: String, cottonPart: Int): List<Socks>
}